### PR TITLE
[receiver/awsecscontainermetrics] Fix possible panics from unchecked de-references

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** Fixes possible panics from unchecked de-references in metrics helper in `awsecscontainermetrics` receiver. 
**Link to tracking Issue:** https://github.com/aws-observability/aws-otel-collector/issues/2110

**Testing:** Added unit tests for these de-references

**Documentation:** n//a

### CODE REVIEW

File: .chloggen/ecsRecvDeRefCheck.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: awsecscontainermetricsreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix possible panics in awsecscontainerrmetrics receiver
```
Comment (Aneurysm9) [inline .chloggen/ecsRecvDeRefCheck.yaml:12]:
```suggestion
note: Fix possible panics in `awsecscontainermetrics` receiver
```

**Review by songy23 — APPROVED** [review]

File: receiver/awsecscontainermetricsreceiver/internal/awsecscontainermetrics/metrics_helper_test.go [hunk 2]
```diff
@@ -95,6 +95,112 @@ func TestGetContainerMetricsMissingMemory(t *testing.T) {
...
+			},
+			prevCpuStats: previousCPUStats,
+		},
+		{
+			memoryStats: mem,
+			testName:    "nil prev cpu stats values",
+			cpuStats:    cpuStats,
+			networkRate: netRate,
+			prevCpuStats: CPUStats{
+				CPUUsage: &CPUUsage{
```
Comment (Aneurysm9) [inline receiver/awsecscontainermetricsreceiver/internal/awsecscontainermetrics/metrics_helper_test.go:168]: Can you add a test where this field is `nil`?  It's not clear that all access to this is safe.

Comment (bryan-aguilar) [reply receiver/awsecscontainermetricsreceiver/internal/awsecscontainermetrics/metrics_helper_test.go:168]: Good catch! You found another one! Gotta catch em all!

Comment (bryan-aguilar) [reply receiver/awsecscontainermetricsreceiver/internal/awsecscontainermetrics/metrics_helper_test.go:168]: Added some additional tests and nil checks!

**Review by Aneurysm9 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**